### PR TITLE
chore: Use git-cliff to manage changelog

### LINKED ISSUE #44: Include a CHANGELOG
Probably with https://github.com/orhun/git-cliff

### PR DESCRIPTION
**Pull request by spencergilbert**
Uses [git-cliff](https://github.com/orhun/git-cliff) to generate the CHANGELOG.md from semantic commits. Needs some automation (either generating from GH actions, or verifying it's been generated), but I'd like to manually figure out the process before trying to script it.

https://github.com/vectordotdev/helm-charts/blob/spencer/changelog/CHANGELOG.md Seems pretty good, at least as a start

Closes #44

### CODE REVIEW

File: .github/cliff.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+commit_parsers = [
+    { message = "^feat", group = "Features"},
+    { message = "^fix", group = "Bug Fixes"},
+    { message = "^doc", group = "Documentation", default_scope = "unscoped"},
+    { message = "^perf", group = "Performance"},
+    { message = "^refactor", group = "Refactor"},
+    { message = "^style", group = "Styling"},
+    { message = "^test", group = "Testing"},
+    { message = "^chore\\(release\\): prepare for", skip = true},
+    { message = "^chore", group = "Miscellaneous Tasks"},
```
Comment (jszwedko) [inline .github/cliff.toml:49]: Maybe these ones should be skipped too since they don't contain user facing changes?

**Review by jszwedko — APPROVED** [review]
I like it

File: .github/cliff.toml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    { body = ".*security", group = "Security"},
+]
+# filter out the commits that are not matched by commit parsers
+filter_commits = false
+# glob pattern for matching git tags
+tag_pattern = "v[0-9]*"
+# regex for skipping tags
+skip_tags = "v0.1.0-beta.1"
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply .github/cliff.toml:57]: 🤔 default config, I'll update that

File: .github/cliff.toml (hunk 1 above)
Comment (spencergilbert) [reply .github/cliff.toml:49]: Yeah, I guess that depends on what we want to include in the changelog, only user facing things seems fair sorta https://keepachangelog.com/ style?

Comment (jszwedko) [reply .github/cliff.toml:49]: Yeah, I think so. It'd help keep the noise down.

Comment (spencergilbert) [reply .github/cliff.toml:49]: I need people to know all the choring I've been doing!

Comment (spencergilbert) [reply .github/cliff.toml:49]: By the same logic - `refactor` could probably be removed as well?

**Review by StephenWakely — APPROVED** [review]
Nice idea.

**Pull request merged**